### PR TITLE
[etcd-cpp-apiv3] new port

### LINKED ISSUE #18901: etcd-cpp-apiv3:x64-windows build failure
**Host Environment**
- OS: windows 10
- Compiler: Vcpkg version: 2021-05-05-e8977e69d9a3fb462d9ad42013d83a7682706659

**To Reproduce**
vcpkg install etcd-cpp-apiv3:x64-windows

**Failure logs** 
D:\third_party\vcpkg>vcpkg install etcd-cpp-apiv3:x64-windows
Computing installation plan...
The following packages will be built and installed:
    etcd-cpp-apiv3[core]:x64-windows -> 0.2.1
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\z\AppData\Local\vcpkg\archives\1d\1d4690500cc8cc25f53c954091805479fe2290c0d7f2edfa855fa73eca80b0e8.zip
Starting package 1/1: etcd-cpp-apiv3:x64-windows
Building package etcd-cpp-apiv3[core]:x64-windows...
-- Using D:/third_party/vcpkg/downloads/etcd-cpp-apiv3-etcd-cpp-apiv3-v0.2.1.tar.gz
-- Cleaning sources at D:/third_party/vcpkg/buildtrees/etcd-cpp-apiv3/src/v0.2.1-85b6180791.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/third_party/vcpkg/downloads/etcd-cpp-apiv3-etcd-cpp-apiv3-v0.2.1.tar.gz
-- Using source at D:/third_party/vcpkg/buildtrees/etcd-cpp-apiv3/src/v0.2.1-85b6180791.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: D:/third_party/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: D:/third_party/vcpkg/buildtrees/etcd-cpp-apiv3/x64-windows-dbg
    See logs for more information:
      D:\third_party\vcpkg\buildtrees\etcd-cpp-apiv3\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/etcd-cpp-apiv3/portfile.cmake:18 (vcpkg_install_cmake)
  scripts/ports.cmake:139 (include)

Error: Building package etcd-cpp-apiv3:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: etcd-cpp-apiv3:x64-windows
  Vcpkg version: 2021-05-05-e8977e69d9a3fb462d9ad42013d83a7682706659

Additionally, attach any relevant sections from the log files above.

**Additional context**
nothing i can find to do

### PR DESCRIPTION
**Pull request by mathisloge**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18901 
the port was not in vcpkg an required to copy the dir of the repo. thought it would be good to have it as a native port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
yes

### CODE REVIEW

File: ports/etcd-cpp-apiv3/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "homepage": "https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3",
+  "dependencies": [
+    "boost-asio",
+    "boost-random",
+    "boost-system",
+    "boost-thread",
+    "cpprestsdk",
+    "grpc",
+    "openssl",
+    "protobuf"
```
Comment (PhoebeHui) [inline ports/etcd-cpp-apiv3/vcpkg.json:14]:
```suggestion
    "protobuf",
     {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

File: ports/etcd-cpp-apiv3/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO etcd-cpp-apiv3/etcd-cpp-apiv3
+    REF v0.2.1
+    SHA512 2112d4f59386e7752f1e7930faf1604c0208dcb7ee8fc431e6cc4c5ce3b4616cd8dacfd68af560d0f973b1b8a49d881d33e29923ec3a4b16a4de1a4f5db270c5
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (PhoebeHui) [inline ports/etcd-cpp-apiv3/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH ${SOURCE_PATH}
```

File: ports/etcd-cpp-apiv3/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    OPTIONS
+        -DBUILD_ETCD_TESTS=OFF
+)
+set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+set(VCPKG_POLICY_DLLS_WITHOUT_EXPORTS enabled)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/etcd-cpp-api)
```
Comment (PhoebeHui) [inline ports/etcd-cpp-apiv3/portfile.cmake:19]:
```suggestion
vcpkg_cmake_install()
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/etcd-cpp-api)
```

File: ports/etcd-cpp-apiv3/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib)
+
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/etcd-cpp-apiv3 RENAME copyright)
```
Comment (PhoebeHui) [inline ports/etcd-cpp-apiv3/portfile.cmake:30]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@mathisloge, thanks for your contribution!

Ping me if this PR is ready for review!

**Comment by mathisloge** [comment]
currently didn't have the time to fix those issues on linux. 

close this for now.

**Pull request closed without merging**